docs/DeveloperGuide.md

```diff
@@ -1,34 +1,890 @@
 
-{Describe the target user profile}
+## 3. Design
+This section provides a high level overview of our application, Kitchen Helper.
+### 3.1. Architecture
 
-### Value proposition
+![Architecture](images/KitchenHelperMain.png)
+The image above explains the design of the application, Kitchen Helper. 
 
-{Describe the value proposition: what problem does it solve?}
+The main driver of the application is `Main: Kitchen Helper`. It is responsible for mainly two phases:
+- At application launch
+    - This class will initialise the components in the correct sequence and is in charge of connecting them with each other.
+- At shut down
+    - This class will invoke cleanup method for the components when necessary.
+    
+In addition to that, the architecture of Kitchen Helper is broken down into seven classes, mainly the following: 
+- `Ui`: This class mainly handles the UI of the application.
+- `Parser`: This class mainly handles the parsing and handling of user commands.
+- `Command`: This class handles the type of command.
+- `Ingredient`: This class manages the data of data type ingredient in memory.
+- `Chore`: This class manages the data of data type chore in memory.
+- `Recipe`: This class manages the data of data type recipe in memory.
+- `Storage`: This class reads data from and writes data back into a text file for future uses.
 
-## User Stories
+### 3.2. Ui Component
+![Ui Component](images/UI_Component.png)
+
+API: `Ui.java`
+ 
+The `Ui` component is a singleton class where all interaction will be made through this component
+ 
+The `Ui` component,
+
+* Executes user commands using the command component
+* Listens for changes and outputs messages from the Command component
+
+### 3.3. Logic Component
+
+![Logic Component](images/logic.png)
+
+1. `Kitchen Helper` uses `Parser` class to parse the user command.
+2. This results in a command object return back which is executed by `Kitchen Helper`.
+3. The command execution can affect the object (e.g. adding an ingredient).
+4. The result of the command execution is encapsulated as a `CommandResult` object which is passed back to `Ui` to display the message.
+
+### 3.4. Model Component
+
+![Model Component](images/model.png)
+* Stores the ingredient data.
+* Stores the recipe data.
+* Stores the chore data.
+
+### 3.5. Storage Component
+
+![Storage Class Diagram](images/Storage.png)
+
+A Storage object is created by the KitchenHelper class to handle the loading and saving of ingredients, recipes and chores data.
+
+The Storage() method acts as a constructor with filepaths to local save files for ingredients, recipes and chores data.
+
+The getIngredientData(), getRecipeData() and getChoreData() methods are used to read saved data from local files into the current session of KitchenHelper. loadingIngredients() and loadingRecipeItems() methods are called in getIngredientData() and getRecipeData() respectively to sort out which Ingredient object class each object belongs to.
```

Comment:
Mentioned above by the fellow 2 reviewers, the diagram above is a bit small with a lot of info. Besides that, perhaps the methods description in this line of code can be written with a special formatting as it is a bit lengthy. E.g, using `getIngredientData()`, `getRecipeData()` and `getChoreData()` instead of getIngredientData(), getRecipeData() and getChoreData()